### ISSUE TITLE
filterwarnings mark escapes regular expressions

### ISSUE THREAD

**Issue by nicoddemus** [pytest-dev/pytest#3936]
The following example should not print any warnings:

```python
import pytest, warnings

@pytest.mark.filterwarnings(r"ignore:some \(warning\)")
def test_foo():
    warnings.warn(UserWarning("some (warning)"))     
```

```
λ pytest .tmp\test-warn-regex.py
======================== test session starts ========================
...
collected 1 item

.tmp\test-warn-regex.py .                                      [100%]

========================= warnings summary ==========================
test-warn-regex.py::test_foo
  c:\pytest\.tmp\test-warn-regex.py:6: UserWarning: some (warning)
    warnings.warn(UserWarning("some (warning)"))

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=============== 1 passed, 1 warnings in 0.02 seconds ================
```

**Comment by asottile** [comment]
This seems to be the same issue as #3785 -- though I see the commit attached to this, I'm having a hard time finding the PR? /me confused

**Comment by nicoddemus** [comment]
My bad, this was solved by #3931. 👍
